src/main/scala/com/codacy/analysis/cli/tools/Tool.scala

```diff
@@ -132,9 +132,20 @@ class Tool(private val plugin: IDockerPlugin) {
 
 object Tool {
 
-  private val allTools = PluginHelper.dockerPlugins.++(PluginHelper.dockerEnterprisePlugins)
+  val allInternetToolShortNames: Set[String] =
+    PluginHelper.dockerEnterprisePlugins.map(_.shortName)(collection.breakOut)
 
-  def allToolShortNames: Set[String] = allTools.map(_.shortName)(collection.breakOut)
+  val allToolShortNames: Set[String] = allInternetToolShortNames ++ PluginHelper.dockerPlugins.map(_.shortName)
+}
+
+class ToolCollector(allowNetwork: Boolean) {
+
+  private val availableTools =
+    PluginHelper.dockerPlugins.++ {
+      if (allowNetwork) {
```

Comment:
Try to not us logic inside other computations. Make the if explicit and then join after.